addons/account_analytic_analysis/account_analytic_analysis.py

```diff
@@ -93,486 +68,497 @@ def product_id_change(self, cr, uid, ids, product, uom_id, qty=0, name='', partn
+    def _hours_qtt_invoiced_calc(self):
+        for account in self:
+            account.hours_qtt_invoiced = account.hours_quantity - account.hours_qtt_non_invoiced
+            if account.hours_qtt_invoiced < 0:
+                account.hours_qtt_invoiced = 0.0
+        for record in self.ids:
+            record.hours_qtt_invoiced = round(record.hours_qtt_invoiced, 2)
+
+    @api.multi
+    @api.depends('hours_qtt_invoiced','ca_invoiced','hours_qtt_invoiced','revenue_per_hour')
+    def _revenue_per_hour_calc(self):
+        for account in self:
             if account.hours_qtt_invoiced == 0:
-                res[account.id]=0.0
+                account.revenue_per_hour = 0.0
             else:
-                res[account.id] = account.ca_invoiced / account.hours_qtt_invoiced
-        for id in ids:
-            res[id] = round(res.get(id, 0.0),2)
-        return res
-
-    def _real_margin_rate_calc(self, cr, uid, ids, name, arg, context=None):
-        res = {}
-        for account in self.browse(cr, uid, ids, context=context):
+                account.revenue_per_hour = account.ca_invoiced / account.hours_qtt_invoiced
+        for record in self.ids:
+            record.revenue_per_hour = round(record.revenue_per_hour,2)
+
+    @api.multi
+    @api.depends('ca_invoiced','total_cost','real_margin','total_cost')
+    def _real_margin_rate_calc(self):
+        for account in self:
             if account.ca_invoiced == 0:
-                res[account.id]=0.0
+                account.real_margin_rate = 0.0
             elif account.total_cost != 0.0:
-                res[account.id] = -(account.real_margin / account.total_cost) * 100
+                account.real_margin_rate = -(account.real_margin / account.total_cost) * 100
             else:
-                res[account.id] = 0.0
-        for id in ids:
-            res[id] = round(res.get(id, 0.0),2)
-        return res
-
-    def _fix_price_to_invoice_calc(self, cr, uid, ids, name, arg, context=None):
-        sale_obj = self.pool.get('sale.order')
-        res = {}
-        for account in self.browse(cr, uid, ids, context=context):
-            res[account.id] = 0.0
-            sale_ids = sale_obj.search(cr, uid, [('project_id','=', account.id), ('state', '=', 'manual')], context=context)
-            for sale in sale_obj.browse(cr, uid, sale_ids, context=context):
-                res[account.id] += sale.amount_untaxed
+                account.real_margin_rate = 0.0
+        for record in self.ids:
+            record.real_margin_rate = round(record.real_margin_rate,2)
+
+    @api.multi
+    def _fix_price_to_invoice_calc(self):
+        sale_obj = self.env['sale.order']
+        for account in self:
+            account.fix_price_to_invoice = 0.0
+            sale_ids = sale_obj.search([('project_id','=', account.id), ('state', '=', 'manual')])
+            for sale in sale_ids:
+                account.fix_price_to_invoice += sale.amount_untaxed
```

Comment:
Be careful to use Active Record Pattern while writing the value. As Each assignment will trigger a write action on the database !!!